Module01/Lexer.py

```diff
@@ -1,288 +1 @@
-                    self.is_alpha = False
-                    return ('VARIABLE', self.current_token)
-            
-            #token 'int or decimal'
-            if (self.current_char.isdigit() or self.current_char == '.'):
-                if not self.past_assign_op:
-                    self.error(3)
-                if self.is_number: # Determines if continueing number or new number.
-                    self.current_token += self.current_char
-                    if(self.current_char == '.'):
-                        self.period_counter +=1
-                else:
-                    self.current_token = self.current_char
-                    self.is_number = True
-                    if(self.current_char == '.'):
-                        self.error(1)  
-            else:
-                if(self.is_number):
-                    if(self.period_counter < 2):
-                        self.period_counter = 0
-                        self.is_number = False
-                        return ('VALUE', self.current_token)
-                    else:
-                        self.error(2)
-                    self.is_number = False
-                    
-            #whitespace to be ignored
-            if self.current_char.isspace():
-                self.advance()
-                return ('SPACE', ' ')
-                #continue
-            
-            #token '+'
-            if self.current_char == '+':
-                if not self.past_assign_op:
-                    self.error(4)
-                self.advance()
-                return ('ADD', '+')
-            
-            #token '-'
-            if self.current_char == '-':
-                if not self.past_assign_op:
-                    self.error(4)
-                self.advance()
-                return ('SUBTRACT', '-')
-            
-            #token '*'
-            if self.current_char == '*':
-                if not self.past_assign_op:
-                    self.error(4)
-                self.advance()
-                return ('MULTIPLY', '*')
-            
-            #token '/'
-            if self.current_char == '/':
-                if not self.past_assign_op:
-                    self.error(4)
-                self.advance()
-                return ('DIVIDE', '/')
-            
-            #token '='
-            if self.current_char == '=':
-                self.advance()
-                self.past_assign_op = True
-                return ('ASSIGN', '=')
-            
-            #check invalid character
-            if not (self.is_alpha or self.is_number):
-                self.error(0)
-
-            self.advance() # advancing for number and variables (done after alternate checks)
-
-        # If end of file is reached and number or variable has not been returned
-        # Return them before ending file.
-        if(self.is_alpha):
-            self.is_alpha = False
-            return ('VARIABLE', self.current_token)
-        if(self.is_number):
```

Comment:
You probably have a bug here, because you don't check for the number of periods in the number. I would test whether a = 0..3 gives you exception. If it doesn't, then you have a bug coming from this code.